File: ios/FishLeague/Utilities/QRValidator.swift
import Foundation

enum QRValidationResult {
    case valid(code: String)
    case invalidFormat
    case missing
}

struct QRValidator {
    static let expectedPrefix = "MAT-"

    static func validate(_ code: String?) -> QRValidationResult {
        guard let code, !code.isEmpty else { return .missing }
        guard code.hasPrefix(expectedPrefix), code.count > expectedPrefix.count else {
            return .invalidFormat
        }
        return .valid(code: code)
    }
}

File: mobile/src/utils/qrValidator.ts
const PREFIX = 'MAT-';

export type QRResult =
  | { valid: true; code: string }
  | { valid: false; reason: 'missing' | 'invalidFormat' };

export function validateQR(code?: string | null): QRResult {
  if (!code) return { valid: false, reason: 'missing' };
  if (!code.startsWith(PREFIX) || code.length <= PREFIX.length) {
    return { valid: false, reason: 'invalidFormat' };
  }
  return { valid: true, code };
}
